### PR TITLE
Empty channel when retrying messages fix

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Fix empty state for channel in `ChannelListView` when retrying messages.

### 🛠 Implementation details

Updating state of channels when they are queried. It should not be possible to have an untracked state for channels that were queried. Now every time a new query is queried, its channelData is updated.

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/175951711-ddfb0823-3f8f-4ce4-8c0f-70a9148da739.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/176039799-cd25c045-1b8a-4102-a083-549f205178f3.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

- Send messages offline
- Kill the app
- Come back
- Enable internet

Check that the channel doesn't get empty (no avatar and the name becomes "channel without name") while messages are being sent.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [] Changelog is updated with client-facing changes
- ~[ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

giphy

### CODE REVIEW

**Comment by leandroBorgesFerreira** [comment]
@kanat No problem, fixing `QueryChannelsLogic ` works too. I just applied the patch.

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt [hunk 1]
```diff
@@ -254,9 +253,9 @@ internal class QueryChannelsLogic(
             logger.logW("Skipping remove channels as they are not loaded yet.")
             return
         }
-        mutableState.queryChannelsSpec.cids = mutableState.queryChannelsSpec.cids - cidList
+        mutableState.queryChannelsSpec.cids = mutableState.queryChannelsSpec.cids - cidList.toSet()
         queryChannelsRepository.insertQueryChannels(mutableState.queryChannelsSpec)
-        mutableState._channels.value = existingChannels - cidList
+        mutableState._channels.value = existingChannels - cidList.toSet()
```
[the remark this answers is not in the document]
Comment (kanat) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt:258]: Makes sense. Good call.
I didn't notice it's also `set` on the left side 👍

Can we then just make `toSet` conversion once?
```kotlin
val cidSet = cidList.toSet()
```

**Review by kanat — APPROVED** [review]
LGTM

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (leandroBorgesFerreira) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt:258]: sure

**Pull request merged**